src/gui/src/device.cpp

```diff
@@ -0,0 +1,2230 @@
+
+devlist::devlist() : QList<device *>() {
+#ifndef QT_NO_DEBUG
+  qDebug("* STA: devlist()");
+#endif
+  devidx = -1;
+#ifndef QT_NO_DEBUG
+  qDebug("* END: devlist()");
+#endif
+};
+
+devlist::~devlist() {
+#ifndef QT_NO_DEBUG
+  qDebug("* STA: ~devlist()");
+#endif
+  clear();
+#ifndef QT_NO_DEBUG
+  qDebug("* END: ~devlist()");
+#endif
+};
+
+int devlist::idx() { return devidx; };
+
+void devlist::setIdx(int iidx) {
+  if (iidx < 0 || iidx >= size()) {
+    devidx = -1;
+    return;
+  }
+  devidx = iidx;
+};
+
+void devlist::clear() {
+  device *dev;
+  while (size()) {
+    dev = takeLast();
+    dev->stopWatcher();
+    delete dev;
+  }
+  devidx = -1;
+};
+
+device *devlist::current() {
+  if (devidx < 0 || devidx >= size())
+    return &NullDev;
+  return (*this)[devidx];
+};
+
+/*
+ *
+ * Device
+ *
+ */
+
+device::device(QObject *p) : QObject(p) {
+#ifndef QT_NO_DEBUG
+  qDebug() << "* STA: device(): " << this << " #" << devcnt++
+           << " parent: " << p;
+#endif
+  preserveMediaInfo = 0;
+  running = 0;
+  type = DevtypeNone;
+  host = "";
+  port = 0;
+  info_set = 0;
+
+  rpc_phase = -1;
+  rpc_reg = -1;
+  rpc_ch = -1;
+  rpc_rst = -1;
+
+  plextor_lock = 0;
+
+  asdb.clear();
+  clearMinfo();
+
+  cap = 0;
+  cap_rd = 0;
+  cap_wr = 0;
+
+  life_dn = -1;
+  features.supported = 0;
+  features.enabled = 0;
+  features.grec = 1.0;
+  features.vrec_cd_pwr = 0;
+  features.vrec_cd_str = 0;
+  features.vrec_dvd_pwr = 0;
+  features.vrec_dvd_str = 0;
+  features.tattoo_inner = 22;
+  features.tattoo_outer = 54;
+
+  features.sm_cd_rd = 0;
+  features.sm_cd_wr = 0;
+  features.sm_dvd_rd = 0;
+  features.psm_cd_rd = 0;
+  features.psm_cd_wr = 0;
+  features.psm_dvd_rd = 0;
+
+  features.as_mode = AS_MODE_AUTO;
+  features.pioq_quiet = PIOQ_STD;
+
+  autoupdate = 0;
+
+  pprocess = 0.0;
+  nprocess = "";
+
+  test_cap = 0;
+  test_req = 0;
+  tests = 0;
+  ctest = 0;
+  WT_simul = 1;
+
+  tspeeds.rt = 1;
+  tspeeds.wt = 1;
+  tspeeds.errc = 1;
+  tspeeds.jb = 1;
+  tspeeds.ft = 1;
+
+  io = new QPxIODevice(this);
+  proc = NULL;
+  sock = NULL;
+  mwatcher = NULL;
+  mutex = new QMutex();
+  io_mutex = new QMutex();
+
+  resReader = new ResultsReader(this);
+  resWriter = new ResultsWriter(this);
+
+  if (!p) {
+#ifndef QT_NO_DEBUG
+    qDebug() << "device: NULL parent!";
+#endif
+  } else {
+    connect(this, SIGNAL(doneMInfo(int)), p, SLOT(mediaUpdated(int)));
+    connect(this, SIGNAL(process_started()), p, SLOT(process_started()));
+    connect(this, SIGNAL(process_finished()), p, SLOT(process_finished()));
+    connect(this, SIGNAL(process_progress()), p, SLOT(process_progress()));
+  }
+  connect(resReader, SIGNAL(finished()), this, SLOT(resLoaderDone()));
+
+#ifndef QT_NO_DEBUG
+  qDebug("* END: device()");
+#endif
+};
+
+device::~device() {
+#ifndef QT_NO_DEBUG
+  qDebug() << "* STA: ~device(): " << this << " #" << --devcnt;
+#endif
+  stopWatcher();
+  if (mwatcher) {
+    delete mwatcher;
+  }
+  QTreeWidgetItem *item;
```

Comment:
Codacy found an issue: [The scope of the variable 'item' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)